gcovr/tests/test_gcovr.py

```diff
@@ -380,7 +401,22 @@ def test_build(
         except Exception as e:  # pragma: no cover
             whole_diff_output += "  " + str(e) + "\n"
             if update_reference:
-                shutil.copyfile(coverage_file, reference_file)
+                if CC_REFERENCE in reference_file:
+                    shutil.copyfile(coverage_file, reference_file)
+                else:
+                    reference_dir = os.path.join("reference", CC_REFERENCE)
+                    try:
+                        os.makedirs(reference_dir)
+                    except FileExistsError:
+                        # directory already exists
+                        pass
```

Comment:
can be replaced with `os.makedirs(reference_dir, exist_ok=True)`

Reply:
Copied this part from several lines above. Will change both locations.